### PR TITLE
Tiny CI improvements

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `isort` config fo the first-party imports; fail build reports on non-success statuses.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 4f8d012f0df5b5ddf33a85fc507fc2262a40f855 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62786/4f8d012f0df5b5ddf33a85fc507fc2262a40f855/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/ci/ci.py [hunk 1]
```diff
@@ -2121,6 +2121,8 @@ def main() -> int:
                     pr_info,
                     dump_to_file=True,
                 )
+                if job_report.status != SUCCESS:
+                    exit_code = 1
```
Comment (maxknv) [inline tests/ci/ci.py:2125]: why do you need this?
I don't see a reason for making POST step red, RUN step is red already in this case

Comment (Felixoid) [reply tests/ci/ci.py:2125]: the case is that the Run is not. I'll take a look there, as we discussed

**Comment by Felixoid** [comment]
I assume it's good to go, so let's go!

**Pull request merged**